### PR TITLE
Azure Hosting Recommendation Improvements

### PR DESCRIPTION
**Pull request by ardalis**
## Summary

Adding separate options for Web Apps and Web Apps for Containers.

### CODE REVIEW

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Rename "Windows Azure" to Microsoft Azure.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: For SEO reasons, it is recommended to use descriptive images. For e.g., Change to `Recommended Migration strategy for on-prem .NET apps to Azure App Service`

And, the `image1-6` has a mention of "VSTS" under CI/CD - rename this to Azure DevOps.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Descriptive "alt" text required in the image for SEO.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Descriptive "alt" text required in the image for SEO.
Rename "VSTS" to "Azure DevOps" in the diagram.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Descriptive "alt" text required in the image for SEO.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Descriptive "alt" text required in the image for SEO.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Descriptive "alt" text required in the image for SEO.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Azure Dev Spaces is an AKS only feature. We shouldn't be writing this here in "App service web apps for containers". IMO, It's best to move it to the AKS section and keep it really brief. For Web Apps, we shouldn't be recommending AKS or Dev Spaces but a brief mention is OK. AKS is covered in-depth in Microservices guidance.

**Review by nishanil — CHANGES_REQUESTED** [review]
@ardalis Added some comments for you to address

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (ardalis) [reply docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Ok - I pulled that image tag from an existing one in the docs but I'll run through and add alt text to all of them.

Comment (ardalis) [reply docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Image updated.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (ardalis) [reply docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Moved to AKS section.

**Comment by ardalis** [comment]
@nishanil should be good to go once you approve.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: AKS section gone? I guess this is due to the merge from the master that had your previous changes. The title isn't correct - Web App for containers is already covered above. What we need here is the AKS and the section of Dev Spaces below.

**Review by nishanil — CHANGES_REQUESTED** [review]

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (ardalis) [reply docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Ok that was a rather overzealous auto-merge. Fixed.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: This section **Web App For Containers** and the following paragraph should be dropped.
> Web App for Containers from Azure App Service lets you bring your own Docker formatted container images and easily deploy and run them at scale with Azure. Web App for Containers supports both Linux and Windows containers and can scale to support large scale implementations. 

This is already covered in the section **App Service Web Apps for Containers**

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (nishanil) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: Probably, Using this _service_ is better than using this _product_?

**Review by nishanil — CHANGES_REQUESTED** [review]
Left 2 comments. We should be good to merge when you fix them.

**Comment by ardalis** [comment]
@nishanil All set.

**Review by nishanil — APPROVED** [review]
Looks good for merge.

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: It was correct before. 

```suggestion
App Service Web Apps offers a fully managed platform optimized for hosting web applications. It's a platform as a service (PaaS) offering that lets you focus on your business logic, while Azure takes care of the infrastructure needed to run and scale the app. Some key features of App Service Web Apps:
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
Azure App Service is the best choice for most web apps. Deployment and management are integrated into the platform, sites can scale quickly to handle high traffic loads, and the built-in load balancing and traffic manager provide high availability. You can move existing sites to Azure App Service easily with an online migration tool, use an open-source app from the Web Application Gallery, or create a new site using the framework and tools of your choice. The WebJobs feature makes it easy to add background job processing to your App Service web app. If you have an existing ASP.NET application hosted on-premises using a local database, there's a clear path to migrate the app to an App Service Web App with an Azure SQL Database (or secure access to your on-premises database server, if preferred).
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
In most cases, moving from a locally hosted ASP.NET app to an App Service Web App is a straightforward process. Little or no modification should be required of the app itself, and it can quickly start to take advantage of the many features that Azure App Service Web Apps offer.
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
In addition to apps that aren't optimized for the cloud, Azure App Service Web Apps are an excellent solution for many simple monolithic (non-distributed) applications, such as many ASP.NET Core apps. In this approach, the architecture is basic and simple to understand and manage:
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
A small number of resources in a single resource group is typically sufficient to manage such an app. Apps that are typically deployed as a single unit, rather than those apps that are made up of many separate processes, are good candidates for this [basic architectural approach](https://docs.microsoft.com/azure/architecture/reference-architectures/app-service-web-app/basic-web-app). Though architecturally simple, this approach still allows the hosted app to scale both up (more resources per node) and out (more hosted nodes) to meet any increase in demand. With autoscale, the app can be configured to automatically adjust the number of nodes hosting the app based on demand and average load across nodes.
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
There are a few scenarios where Web Apps for Containers make the most sense. If you have existing apps that you can containerize, whether in Windows or Linux containers, you can host these easily using this toolset. Simply publish your container and then configure Web Apps for Containers to pull the latest version of that image from your registry of choice. This is a "lift and shift" approach to migrating from classic app hosting models to a cloud-optimized model.
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
Developing with containers offers many advantages, especially when containers are used in production. The same container configuration is used to host the app in each environment in which it runs, from local development machine to build and test systems to production. This greatly reduces the likelihood of defects resulting from differences in machine configuration or software versions. Developers can also use whatever tools they're most productive with, including operating system, since containers can run on any OS. In some cases, distributed applications involving many containers may be very resource-intensive to run on a single development machine. In this scenario, it may make sense to upgrade to using Kubernetes and Azure Dev Spaces, covered in the next section.
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/f/front-end
https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/b/back-end

```suggestion
As portions of larger applications are broken up into their own smaller, independent *microservices*, additional design patterns can be used to improve app behavior. Instead of working directly with individual services, an *API gateway* can simplify access and decouple the client from its back end. Having separate service back ends for different front ends also allows services to evolve in concert with their consumers. Common services can be accessed via a separate *sidecar* container, which might include common client connectivity libraries using the *ambassador* pattern.
```

File: docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md
Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]: should probably update ms.date on this article as well (metadata on top)

Comment (mairaw) [inline docs/standard/modern-web-apps-azure-architecture/azure-hosting-recommendations-for-asp-net-web-apps.md]:
```suggestion
> "Line-of-business leaders everywhere are bypassing IT departments to get applications from the cloud (also known as SaaS) and paying for them like they would a magazine subscription. And when the service is no longer required, they can cancel the subscription with no equipment left unused in the corner."  
```

**Comment by ardalis** [comment]
:shipit:

**Review by mairaw — CHANGES_REQUESTED** [review]
Some very quick edit comments. There are more style and voice recommendations from Acrolinx that perhaps @nishanil can help with later on.

**Comment by ardalis** [comment]
should be good now

**Review by mairaw — APPROVED** [review]

**Pull request merged**
